From cce5ccd08b68b544db2409cc5c02a58b49da414c Mon Sep 17 00:00:00 2001
From: Daniel Baluta <daniel.baluta@nxp.com>
Date: Tue, 13 Nov 2018 15:21:36 +0200
Subject: [PATCH 5122/5242] MLK-20189-10: ASoC: fsl: dsp: Avoid unnecessary
 check

commit  de2f62c6a59c21523468425bdfb66fa6734aa92f from
https://source.codeaurora.org/external/imx/linux-imx.git

strtab is always non-null so remove unnecessary check.

This is a follow up of patch
167a6d79f ("ASoC: fsl: Skip checking for string section type")
and finally fixes CID3901026

Reviewed-by: Shengjiu Wang <shengjiu.wang@nxp.com>
Signed-off-by: Daniel Baluta <daniel.baluta@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 sound/soc/fsl/fsl_dsp.c |   11 ++++-------
 1 file changed, 4 insertions(+), 7 deletions(-)

diff --git a/sound/soc/fsl/fsl_dsp.c b/sound/soc/fsl/fsl_dsp.c
index b251a18..e2ccf9c 100644
--- a/sound/soc/fsl/fsl_dsp.c
+++ b/sound/soc/fsl/fsl_dsp.c
@@ -669,13 +669,10 @@ static void dsp_load_firmware(const struct firmware *fw, void *context)
 			shdr->sh_addr == 0 || shdr->sh_size == 0)
 			continue;
 
-		if (strtab) {
-			dev_dbg(dev, "%sing %s @ 0x%08lx (%ld bytes)\n",
-			  (shdr->sh_type == SHT_NOBITS) ? "Clear" : "Load",
-				&strtab[shdr->sh_name],
-				(unsigned long)shdr->sh_addr,
-				(long)shdr->sh_size);
-		}
+		dev_dbg(dev, "%sing %s @ 0x%08lx (%ld bytes)\n",
+			(shdr->sh_type == SHT_NOBITS) ? "Clear" : "Load",
+			&strtab[shdr->sh_name], (unsigned long)shdr->sh_addr,
+			(long)shdr->sh_size);
 
 		sh_addr = shdr->sh_addr;
 
-- 
1.7.9.5

